### DISCUSSION TITLE
Discrepancy in logs w.r.t learning rate

### DISCUSSION THREAD

**Discussion by rezafuru** [yoshitomo-matsubara/torchdistill#242] [Q&A] [answered]
Hi,

I've been running some experiments and noticed that the logs give me a different learning rate then what I specified in the config. For example:
```
    optimizer:
      type: 'Adam'
      params:
        lr: 0.001
    scheduler:
      type: 'MultiStepLR'
      params:
        milestones: [6, 9]
        gamma: 0.1
```

Yields `2022/08/20 14:11:44	INFO	torchdistill.misc.log	Epoch: [2]  [100/782]  eta: 0:01:08  lr: 1e-05` in the logs. Stepping through the debugger, it does seem like the optimizer was correctly initiallized. However, the training box seems to store a different lr in its param groups, which is what is logged in the end (`metric_logger.update(loss=loss.item(), lr=training_box.optimizer.param_groups[0]['lr'])`)

Does anyone know whether I'm misunderstanding the configuration, or is this a potential bug? Note, regardless of what I set the lr in the config, the logger keeps outputting `lr: 1e-05`

**Comment by yoshitomo-matsubara** [comment, marked as answer]
Hi @rezafuru 

If you're setting module-wise learning rates, it is possible that you see different learning rate as part of the log since the log is designed to show only one learning rate as you found in `lr=training_box.optimizer.param_groups[0]['lr']`

There is also a chance that you are loading checkpoint (its file path defined in the yaml file) including parameters of optimizer, which won't be seen when instantiating the optimizer, but can be confirmed at the beginning of training like [this](https://github.com/yoshitomo-matsubara/torchdistill/blob/main/examples/image_classification.py#L116). So make sure that you don't load checkpoint for the model to be trained.

If it is not the case, could you also provide more details of your script? If you just reuse one of my scripts under `examples/` in torchdistill, it's easy to debug for me. But if you either edited them or wrote your own script, there is a fair chance to introduce such discrepancies, and I need to check the script to see what's happening.
e.g., did you configure all the components in training_box only by the yaml file? (I also want to take a look at the yaml file)

**Reply by rezafuru** [reply]
Hello, thank you for your quick response. Indeed, the problem was that I was loading a model from a checkpoint.
